controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -180,37 +213,73 @@ public String getName() {
+                                    List<String> streamTagsInScopeTableNames = streamTagsInScopeTableNameFuture.join();
+                                    val deleteTagTablesFut = streamTagsInScopeTableNames.stream()
+                                                                                        .map(tableName -> storeHelper.deleteTable(tableName, false, context.getRequestId()))
+                                                                                        .collect(Collectors.toList());
+                                    return CompletableFuture.allOf(storeHelper.deleteTable(streamsInScopeTableName, true, context.getRequestId()),
+                                                                   storeHelper.deleteTable(kvtsInScopeTableName, true, context.getRequestId()),
+                                                                   storeHelper.deleteTable(rgsInScopeTableName, true, context.getRequestId()),
+                                                                   Futures.allOf(deleteTagTablesFut))
+                                                            .thenAccept(v -> log.debug("tables deleted {} {} {}", streamsInScopeTableName,
+                                                                                       kvtsInScopeTableName, rgsInScopeTableName));
+                                })
+                                .thenCompose(deleted -> storeHelper.removeEntry(SCOPES_TABLE, scopeName, context.getRequestId()));
     }
 
     @Override
-    public CompletableFuture<Pair<List<String>, String>> listStreams(int limit, String continuationToken, Executor executor, 
+    public CompletableFuture<Pair<List<String>, String>> listStreams(int limit, String continuationToken, Executor executor,
                                                                      OperationContext context) {
         Preconditions.checkNotNull(context, "Operation context cannot be null");
         return getStreamsInScopeTableName(context)
                 .thenCompose(streamsInScopeTable -> readAll(limit, continuationToken, streamsInScopeTable, context));
     }
 
+    @Override
+    public CompletableFuture<Pair<List<String>, String>> listStreamsForTag(String tag, String continuationToken, Executor executor,
+                                                                           OperationContext context) {
+
+        Preconditions.checkNotNull(context, "Operation context cannot be null");
+
+        return getNextTableSegment(tag, continuationToken, context).thenCompose(pair -> {
+            if (pair.getLeft().isEmpty() && !pair.getRight().contains(LAST_TAG_CHUNK)) {
+                return listStreamsForTag(tag, pair.getRight(), executor, context);
+            } else {
+                return CompletableFuture.completedFuture(pair);
+            }
+        });
+    }
+
+    CompletableFuture<Pair<List<String>, String>> getNextTableSegment(String tag, String ct, OperationContext context) {
```

Comment:
the method name says get next table segment. and it returns a pair - list of strings and a continuation token. 
could you please add javadoc explaining what this is doing here. i believe it is returning the list of stream names from the continuation token, right?

Reply:
fixed it.